### PR TITLE
it works

### PR DESCRIPTION
**Pull request by yurko**
Hi, deadlyicon! I really like your idea to backup pins, so made some fixes to script.
- scrolling is fixed;
- get around with pinterest's hiding of top pins when scrolled down;
- change images' links to wider 736px instead of 236px.

Now it works.. at least on my system :)
Isuues:
- failes to load big boards, in my case only ~1400 pins of 4600 in board;
- sometimes fails even with small boards, loads only ~25 pins. Maybe need more "sleep"

### CODE REVIEW

File: backup.rb [hunk 1]
```diff
@@ -66,41 +71,56 @@ def get_images!
     @boards.each do |board|
       print "[#{board[:name]}] ".green + "found "
       visit PINTEREST + board[:path]
-      load_all_content!
-      board[:image_urls] = js <<-JS
-        $('.pin').map(function(){ return $(this).data('closeup-url') });
-      JS
+      sleep 3
```
Comment (deadlyicon) [inline backup.rb]: we shouldn't have to sleep

File: backup.rb [hunk 2]
```diff
@@ -66,41 +71,56 @@ def get_images!
...
       puts "#{board[:image_urls].size} images"
+
       download_images! board
     end
   end
 
+  def get_unique_links!
+    image_links = []
+
+      until scrolled_to_bottom?
```
Comment (deadlyicon) [inline backup.rb]: indentation error here.

File: backup.rb [hunk 3]
```diff
@@ -66,41 +71,56 @@ def get_images!
...
+  end
+
   def download_images! board
     path = @root + 'pinterest.com' + board[:path][1..-1]
     image_urls = board[:image_urls].map(&:to_s).map(&:inspect).join(' ')
     path.mkpath
     return if image_urls.empty?
     command = <<-SH
-      cd #{path.to_s.inspect} && wget --quiet --background #{image_urls}
+      cd #{path.to_s.inspect} && wget -c --background #{image_urls}
```
Comment (deadlyicon) [inline backup.rb]: i like the adding of `-c` but why remove `--quiet`?

File: backup.rb [hunk 4]
```diff
@@ -66,41 +71,56 @@ def get_images!
...
     SH
-    # puts command
     system command
   end
 
   def load_all_content!
-    while !scrolled_to_bottom?
```
Comment (deadlyicon) [inline backup.rb]: these changes break this function. it will no longer load all content.

File: backup.rb [hunk 5]
```diff
@@ -66,41 +71,56 @@ def get_images!
...
       js %(window.scrollTo(0, 9999999))
-      sleep 1
-    end
+      sleep 3
   end
 
   def scrolled_to_bottom?
     js <<-JS
-      $('html').height() - document.body.scrollTop <= $(window).height()
+      $(window).scrollTop() == $(document).height() - $(window).height()
```
Comment (deadlyicon) [inline backup.rb]: why did this need to change?

Comment (yurko) [inline backup.rb]: Was problem with scrolling. I just make fixes to make script works for me.
It's your code so consider by yourself what will be usefull for
functionality

Пн, 4 трав. 2015 01:46 Jared Grippe <notifications@github.com> пише:

File: backup.rb (hunk 4 above)
Comment (yurko) [inline backup.rb]: everything works. This loop is changed to:  
  until scrolled_to_bottom?
    load_all_content!
    ...
  end
in get_unique_links! method

File: backup.rb (hunk 3 above)
Comment (yurko) [inline backup.rb]: just to see that wget is running

File: backup.rb (hunk 2 above)
Comment (yurko) [inline backup.rb]: yep.. but it doesn't affect functionality

File: backup.rb (hunk 1 above)
Comment (yurko) [inline backup.rb]: by testing came to conclusion that need sleep to page get fully loaded before js evaluation

**Pull request closed without merging**